src/daft-io/src/s3_like.rs

```diff
@@ -1618,6 +1606,21 @@ impl S3MultipartWriter {
     }
 }
 
+#[async_trait]
+impl MultipartWriter for S3MultipartWriter {
```

Comment:
Just reused the existing logic, but the S3MultipartWriter could be refactored, which can depends S3 client directly to implement mpu API instead of S3LikeSource.

will submit another PR to refactor it if the current design is acceptable.